src/sentry/api/serializers/models/incident.py

```diff
@@ -97,5 +98,19 @@ def serialize(self, obj, attrs, user):
         context["hasSeen"] = seen_list["has_seen"]
         context["groups"] = attrs["groups"]
         context["alertRule"] = serialize(obj.alert_rule, user)
+        # The query we should use to get accurate results in Discover.
+        context["discoverQuery"] = self._build_discover_query(obj)
 
         return context
+
+    def _build_discover_query(self, incident):
+        query = incident.query
+        if (
+            incident.alert_rule
+            and QueryDatasets(incident.alert_rule.dataset) == QueryDatasets.EVENTS
+        ):
+            query = incident.alert_rule.query
+            condition = "event.type:error"
+            query = "{} {}".format(condition, query) if query else condition
```

Comment:
I guess this is not intended to be perfect since it's mostly intended for a short term patch? I'm wondering about cases where this added condition would conflict with something already in the query and lead to strange results for the user.

Reply:
Looks good otherwise.

Reply:
If the user has another `event.type` condition you'll get a query that will yield 0 results, but otherwise this should be safe.

Reply:
Yeah, that's a potential problem, but that is what is going to happen in the subscription as well. Discussion here: https://app.asana.com/0/1143846759074121/1170558951694547